### PR TITLE
Removes FileEditorCodeWrapper

### PR DESCRIPTION
**Pull request by hbcarlos**
## References
Follow up:
- #12338
- #12381

## Code changes
Removes the FileEditorCodeWrapper class.

## Backwards-incompatible changes
Removes the class FileEditorCodeWrapper, instead, you can use the CodeEditorWrapper from `@jupyterlab/codeeditor`.

### CODE REVIEW

File: packages/fileeditor/src/widget.ts [hunk 1]
```diff
@@ -242,11 +161,12 @@ export class FileEditor extends Widget {
     );
   }
 
-  private editorWidget: FileEditorCodeWrapper;
+  private editorWidget: CodeEditorWrapper;
   public model: CodeEditor.IModel;
   public editor: CodeEditor.IEditor;
   protected _context: DocumentRegistry.Context;
   private _mimeTypeService: IEditorMimeTypeService;
+  private _ready = new PromiseDelegate<void>();
```
Comment (fcollonval) [inline packages/fileeditor/src/widget.ts:169]: Let's fix the visibility (`context` has public getter so no need to have `_context` protected as it should not be set outside the constructor) and follow the naming convention (`editorWidget` -> `_editorWidget`):

```suggestion
  model: CodeEditor.IModel;
  editor: CodeEditor.IEditor;
  private _context: DocumentRegistry.Context;
  private _editorWidget: CodeEditorWrapper;
  private _mimeTypeService: IEditorMimeTypeService;
  private _ready = new PromiseDelegate<void>();
```

**Review by fcollonval — CHANGES_REQUESTED** [review]
Thanks @hbcarlos 

Would you mind applying the suggestion and adding a note in the [migration doc](https://github.com/jupyterlab/jupyterlab/blob/master/docs/source/extension/extension_migration.rst)

**Comment by hbcarlos** [comment]
Thanks for the review @fcollonval! I applied your suggestions, let me know if it is ok.

**Review by fcollonval — APPROVED** [review]
Thanks @hbcarlos

**Comment by jtpio** [comment]
Thanks!

**Pull request merged**